### PR TITLE
feat(Core/Hooks): add few new hooks

### PR DESCRIPTION
**Pull request by Err0r1**
## Changes Proposed:
-  Add new hooks:
`void OnPlayerHonorChange(Player* player, int32& amount);`
`void OnPlayerArenaPointsChange(Player* player, int32& amount);`
`void OnBeforeLootItem(Player* player, LootItem* item, uint64 lootguid);`
`void OnLootGold(Player* player, uint32& amount);`

## Issues Addressed:
- Closes 

## Tests Performed:

- Compiled on debian 10
- 

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@Err0r1 thanks for your contribution! please fix the merge conflicts

**Comment by Yehonal** [comment]
Thanks for your contribution!

Please could you document your hooks (within the ScriptMgr.h file) using the following guide?

https://www.azerothcore.org/wiki/how-to-document-code

IMPORTANT: In future we will setup a doxygen page where anyone can browse the hooks documentation. But for now having a proper documentation within your IDE intellisense or just letting people reading that file to find the proper hook, is already a good improvement.

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -7579,6 +7579,9 @@ void Player::SetArenaPoints(uint32 value)
 
 void Player::ModifyHonorPoints(int32 value, SQLTransaction* trans /*=nullptr*/)
 {
+
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Player/Player.cpp:7582]: no need line

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -7595,6 +7598,9 @@ void Player::ModifyHonorPoints(int32 value, SQLTransaction* trans /*=nullptr*/)
 
 void Player::ModifyArenaPoints(int32 value, SQLTransaction* trans /*=nullptr*/)
 {
+
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Player/Player.cpp:7601]: no need line

**Comment by temperrr** [comment]
@Err0r1 

Hey! I am checking some older PR's to check on their status. What's the status of this PR, is it WIP?

**Pull request closed without merging**